lib/system/extra/cmd_streamer.ex

```diff
@@ -0,0 +1,143 @@
+defmodule System.Extra.CmdStreamer do
+  @moduledoc """
+  Module allowing Enumerables to be constructed around Shell commands.
+
+  """
+
+  alias __MODULE__
+
+
+
+  #########################################################
+  # Types
+  #########################################################
+
+  @opaque t :: %__MODULE__{
+    port: port,
+    queue: :queue.queue,
+    receiver: pid,
+    running: boolean,
+    pending_demand: non_neg_integer,
+  }
+
+  defstruct [
+    port: nil,
+    queue: :queue.new(),
+    receiver: nil,
+    running: true,
+    pending_demand: 0,
+  ]
+
+
+
+  #########################################################
+  # Public API
+  #########################################################
+
+  @doc """
+  Starts the `CmdStreamer` enumerable by wrapping `cmd` and forwarding `args` to it.
+
+  """
+  @spec start(binary, [binary]) :: pid
+  def start(cmd, args) do
+    receiver =
+      self()
+
+    cmd_string =
+      [cmd | args] |> Enum.join(" ")
+
+    spawn(fn ->
+      port =
+        Port.open({:spawn, cmd_string}, [{:line, 20_000}, :binary, :exit_status])
+
+      state =
+        %CmdStreamer{port: port, receiver: receiver}
+
+      loop(state)
+    end)
+  end
+
+
+  @doc """
+  Requests the most recent output from the `CmdStreamer`.
+
+  """
+  @spec get_line(pid) :: {:line, binary} | {:error, non_neg_integer} | :done
+  def get_line(pid) do
+    send(pid, {self(), :next_line})
+    receive do
+      {^pid, msg} -> msg
+    end
+  end
+
+
+
+  #########################################################
+  # Private Helpers
+  #########################################################
+
+  @spec loop(t) :: no_return
+  defp loop(%CmdStreamer{receiver: receiver, port: port} = state) do
+    receive do
+      {^port, {:data, {:eol, line}}} ->
```

Comment:
Hmm, so is this the part where it breaks on 20,000+ character lines? Since they would be delivered with :noeol there would be no handling here. Does it fail silently or does it cause the process to die?